httpobs/scanner/analyzer/headers.py

```diff
@@ -4,6 +4,50 @@
 from httpobs.scanner.analyzer.utils import is_hpkp_preloaded, is_hsts_preloaded, only_if_worse
 
 
+def __parse_csp(csp_string: str) -> dict:
+    """
+    Decompose the CSP; could probably do this in one step, but it's complicated enough
+    Should look like:
+    {
+      'default-src': {'\'none\''},
+      'object-src': {'\'none\''},
+      'script-src': {'https://mozilla.org', '\'unsafe-inline\''},
+      'style-src': {'\'self\', 'https://mozilla.org'},
+      'upgrade-insecure-requests': {},
+    }
+    """
+
+    # Clean out all the junk
+    csp_string = csp_string.replace('\r', '').replace('\n', '').strip()
+
+    # So technically the shortest directive is img-src, so lets just assume that
+    # anything super short is invalid
+    if len(csp_string) < 6 or csp_string.isspace():
+        raise ValueError
```

Comment:
I don't like the use of a magic number here. I'd suggest defining it as a constant, and comparing against

```python
SHORTEST_DIRECTIVE = 'img-src'
SHORTEST_DIRECTIVE_LENGTH = len(SHORTEST_DIRECTIVE)
```

Then in addition to having more obvious and legible code, you can import these and use them as comparators for tests.

Reply:
Fixed, thank you.